src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -506,3 +519,25 @@ invariant
     -> a
 invariant msg a predicate =
     if predicate a then a else error msg
+
+
+-- | Magic number which should differ for different clusters. It's
+-- defined here, because it's used for signing. It also used for other
+-- things (e. g. it's part of a serialized block).
+--
+-- mhueschen: As part of CO-353 I am adding `getRequiresNetworkMagic` in
+-- order to pipe configuration to functions which must generate & verify
+-- Addresses (which now must be aware of `NetworkMagic`).
+newtype ProtocolMagic = ProtocolMagic
+    { getProtocolMagicId      :: Int32
+    --, getRequiresNetworkMagic :: !RequiresNetworkMagic
+    } deriving (Eq, Show, Generic)
+
+{-# NOINLINE protocolMagic #-}
+protocolMagic :: Int32
+protocolMagic = fromMaybe testnet (envvar >>= readMaybe)
+  where
+    testnet = error "what /is/ the testnet pm?"
```

Comment:
mainnet: 764824073
testnet: 1097911063

Thinking a second time aobut this, maybe we can actually default to `mainnet` :thinking: ? What do you think ?

Reply:
🤔 Wouldn't it be "safest" to have no default? Or otherwise default to testnet? Such that `testnet` txs don't accidentally end up on mainnet?

Or would defaulting to `mainnet` give us fewer worries in production?

Reply:
Yeah right. Crashing with a clear message that we expect an ENV var with an appropriate protocol magic. I'd even go further and not accept _any_ value, but only the ones we recognize (adding staging: 633343913).